packages/connectors/ledger-connector/src/hid/connector.ts

```diff
@@ -1,116 +1,103 @@
 import {
-  Address,
-  Chain,
-  Connector,
-  ConnectorData,
+  ChainNotConfiguredError,
   ConnectorNotFoundError,
+  createConnector,
 } from 'wagmi';
+import { Chain } from 'wagmi/chains';
 import { checkError } from '../hid/helpers';
+import type { LedgerHQProvider } from './provider';
 
-export class LedgerHIDConnector<Options = any> extends Connector {
-  readonly id = 'ledgerHID';
-  readonly name = 'Ledger';
-  readonly #chain?: Chain;
-  ready = false;
-  #provider?: any;
-  #rpc: string;
+export const idLedgerHid = 'ledgerHID';
+export const name = 'Ledger';
 
-  constructor({
-    defaultChain,
-    chains,
-    rpc,
-    options,
-  }: {
-    defaultChain: Chain;
-    chains?: Chain[];
-    rpc: Record<number, string>;
-    options?: Options;
-  }) {
-    super({ chains, options });
-    this.#chain = defaultChain;
-    this.#rpc = rpc?.[defaultChain.id];
-  }
+export const ledgerHIDConnector = ({
+  rpc,
+  defaultChain,
+}: {
+  rpc: Record<number, string>;
+  defaultChain: Chain;
+}) => {
+  const providers: Record<Chain['id'], LedgerHQProvider> = {};
 
-  async connect(): Promise<Required<ConnectorData>> {
-    try {
-      this.#provider = await this.getProvider();
-      this.#provider.on('disconnect', this.onDisconnect);
-      const account = await this.#provider.enable();
-      const chainId = await this.getChainId();
-      const chainUnsupported = this.isChainUnsupported(chainId);
-      this.ready = !!this.#provider;
+  return createConnector<LedgerHQProvider>(({ chains, emitter }) => ({
+    id: idLedgerHid,
+    name,
+    type: 'ledgerHID', // TODO: define type
```

Comment:
Any updates? I'm afraid that this comment will be here forever, so I think it must be done, or the comment should be removed because it is not really helpful – I don't understand what exactly must be done here.

Reply:
Yes, as I can see from another wagmi connectors this is something like unique identifier for the connector. I have updated this definition pattern to match another connectors from wagmi.